stable/database/files/nuosm

```diff
@@ -104,6 +114,13 @@ function perform_restore() {
     echo "curl -k --user '**' $restore_source | tar xzf - --strip-components $strip_levels -C $DB_DIR"
     curl -k --user "$restore_credentials" $restore_source | tar xzf - --strip-components $strip_levels -C $DB_DIR
 
+    retval=$?
```

Comment:
We are missing handling of the restore type being a backupset _or_ stream in this download - I know that was not the purpose of this PR but we need to cover it off at some point!

Reply:
I've merged the stream/backupset restore code into `perform_restore()` to ensure all restores are handled consistently.